### DISCUSSION TITLE
How do I prime a lockfile with prepinned versions?

### DISCUSSION THREAD

**Discussion by ianjosephwilson** [juftin/hatch-pip-compile#86] [Q&A] [answered]
I already have a requirements.txt created with `pip freeze`  and then hand maintained.  Can I just copy it into where the lockfile should be?  Is that the right way?  I couldn't find it in the docs which probably means it is either too obvious or I probably missed it.  Thanks.

**Comment by juftin** [comment]
This should be much easier than it currently is, I'll look into fixing this and making it a more straightforward process. 

For now, you can place your existing `requirements.txt` where it should be but add the following header to the top of it:

```txt
#
# This file is autogenerated by hatch-pip-compile with Python 3.11
#
# - this-header-will-be-re-written
#
```

The following error is currently raised when it tries and fails to read a non-existent header:

```
LockFileError: Could not find lock file python version
```

**Reply by juftin** [reply, marked as answer]
https://github.com/juftin/hatch-pip-compile/pull/87

Okay, the above issue will be fixed shortly. Just put your pre-existing lockfile where it should be and `hatch-pip-compile` will re-generate it (but also do the right thing by respecing your existing pins)

**Reply by ianjosephwilson** [reply]
Thanks for looking into this so quickly.  Are requirements files manually generated via `pip-compile` compatible with `hatch-pip-compile`?  Or will the difference in the header cause problems?

**Reply by juftin** [reply]
Hey @ianjosephwilson - requirements files generated by `pip-compile` are compatible with `hatch-pip-compile`, but `hatch-pip-compile` will overwrite them to include the header. Assuming your `pip-compile` generated file was up to date though it won't be changed apart from the header being written (similar to how running `pip-compile` back to back doesn't change the lockfile).

**Reply by juftin** [reply]
Also, #87 was merged and the previous mentioned fix was released on [v1.11.3](https://github.com/juftin/hatch-pip-compile/releases/tag/v1.11.3)
